client/src/app.tsx

```diff
@@ -119,18 +126,6 @@ export function App(appProps) {
     document.documentElement.setAttribute("lang", locale);
   }, [appProps.locale, localeMatch]);
 
-  const [pageNotFound, setPageNotFound] = React.useState<boolean>(
-    appProps.pageNotFound
-  );
-  const { pathname } = useLocation();
-  const initialPathname = React.useRef(pathname);
-
-  React.useEffect(() => {
```

Comment:
I guess we don't need the useEffect because `pageNotFound` is inexpensive to calculate?

Reply:
The impetus was because `useEffect` was messing with the types, and yes - indeed - that `pageNotFound` is a straightforward calculation, it requires no "internal" state. I'll wrap it in a `useMemo` though, just to ensure there's no performance regression, since this does get called on practically every render across the entire site.